glnx-console.c

```diff
@@ -280,6 +280,32 @@ glnx_console_progress_text_percent (const char *text,
   text_percent_internal (text, percentage);
 }
 
+/**
+ * glnx_console_progress_n_items:
+ * @text: Show this text before the progress bar
+ * @current: An integer for how many items have been processed
+ * @total: An integer for how many items there are total
+ *
+ * On a tty, print to the console @text followed by [@current/@total],
+ * then an ASCII art progress bar, like glnx_console_progress_text_percent().
+ *
+ * You must have called glnx_console_lock() before invoking this
+ * function.
+ */
+void
+glnx_console_progress_n_items (const char     *text,
+                               guint           current,
+                               guint           total)
+{
+  g_return_if_fail (current <= total);
+  g_return_if_fail (total > 0);
+
+  g_autofree char *newtext = g_strdup_printf ("%s (%u/%u)", text, current, total);
```

Comment:
Minor, though I think it'd look nice if we space/zero pad `current` to be the same size as `total`.

Reply:
Like `Installing ( 32/207)` ?  Is there something else that does that?  What's the goal?  Avoid having the output shift right?  Also offhand I'm not sure how to do that with `printf` without calculating the digits out of band.

Reply:
Right, to avoid the progress bar shifting to the right. `dnf` and `yum` don't do that, but they do right-align the progress bar & speed info and truncate the left text if necessary to make the progress bar stay in the same position. Anyway, we can add that sort of eye candy later if we start e.g. printing each pkg being processed.

Reply:
I was briefly looking for inspiration and saw e.g. https://github.com/a8m/pb but...yeah.

Hmm, I'm not seeing that spacing with f27's dnf at least; 

download:
`(76/262): texlive-avantgar-svn31835.0-36.fc27.5.noarch.rpm                                                                                                                         3.0 MB/s | 310 kB     00:00 `

install:
`  Installing       : texlive-rotating-6:svn16832.2.16b-36.fc27.5.noarch                                                                                                                                    233/262`

Which is pretty weird actually that they're so different.

Reply:
Anyways I pushed this for now since we can always revisit the rendering later, IMO this is a notable improvement.